builtin/providers/openstack/resource_openstack_networking_router_route_v2.go

```diff
@@ -0,0 +1,216 @@
+package openstack
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/networking/v2/extensions/layer3/routers"
+)
+
+func resourceNetworkingRouterRouteV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNetworkingRouterRouteV2Create,
+		Read:   resourceNetworkingRouterRouteV2Read,
+		Delete: resourceNetworkingRouterRouteV2Delete,
+
+		Schema: map[string]*schema.Schema{
+			"region": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OS_REGION_NAME", ""),
+			},
+			"router_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination_cidr": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"next_hop": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceNetworkingRouterRouteV2Create(d *schema.ResourceData, meta interface{}) error {
+
+	routerId := d.Get("router_id").(string)
+	osMutexKV.Lock(routerId)
+	defer osMutexKV.Unlock(routerId)
+
+	var destCidr string = d.Get("destination_cidr").(string)
+	var nextHop string = d.Get("next_hop").(string)
+
+	d.SetId(fmt.Sprintf("%s-route-%s-%s", routerId, destCidr, nextHop))
```

Comment:
It's best to wait to `SetId` until the resource has been confirmed to be created.